test/Sentry.DiagnosticSource.Tests/SentryDiagnosticSubscriberTests.cs

```diff
@@ -0,0 +1,114 @@
+#if !net461
+using System.Collections.Concurrent;
+using System.Data.Common;
+using System.Runtime.InteropServices;
+using LocalDb;
+using Sentry.Internals.DiagnosticSource;
+
+[UsesVerify]
+public class SentryDiagnosticSubscriberTests
+{
+    private static SqlInstance sqlInstance = new(
+            name: "SentryDiagnosticSubscriber",
+            buildTemplate: TestDbBuilder.CreateTable);
+
+    [Fact]
+    public async Task RecordsSql()
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return;
+        }
+        var transport = new RecordingTransport();
+        var options = new SentryOptions
+        {
+            TracesSampleRate = 1,
+            Transport = transport,
+            Dsn = DsnSamples.ValidDsnWithoutSecret,
+            DiagnosticLevel = SentryLevel.Debug
+        };
+
+        using var database = await sqlInstance.Build();
+        options.AddIntegration(new SentryDiagnosticListenerIntegration());
+        using (SentrySdk.Init(options))
```

Comment:
Can we new up a Hub here instead of using the static api?

Reply:
done